src/services/spr.ts

```diff
@@ -0,0 +1,263 @@
+import express from 'express';
+import subdomain from 'express-subdomain';
+import Dicer from 'dicer';
+import { getDuplicateCECData, getRandomCECData } from '@/database';
+import { getNEXDataByPID, getFriends } from '@/util';
+import { CECData } from '@/models/cec-data';
+import { SendMode, SPRSlot } from '@/types/common/spr-slot';
+
+const spr = express.Router();
+
+function multipartParser(request: express.Request, response: express.Response, next: express.NextFunction): void {
+	const RE_BOUNDARY = /^multipart\/.+?(?:; boundary=(?:(?:"(.+)")|(?:([^\s]+))))$/i;
+	const RE_FILE_NAME = /name="(.*)"/;
+
+	const contentType = request.header('content-type');
+
+	if (!contentType) {
+		return next();
+	}
+
+	const boundary = RE_BOUNDARY.exec(contentType);
+
+	if (!boundary) {
+		return next();
+	}
+
+	const dicer = new Dicer({ boundary: boundary[1] || boundary[2] });
+	const files: Record<string, Buffer> = {};
+
+	dicer.on('part', (part: Dicer.PartStream) => {
+		let fileBuffer = Buffer.alloc(0);
+		let fileName = '';
+
+		part.on('header', header => {
+			const contentDisposition = header['content-disposition' as keyof object];
+			const regexResult = RE_FILE_NAME.exec(contentDisposition);
+
+			if (regexResult) {
+				fileName = regexResult[1];
+			}
+		});
+
+		part.on('data', (data: Buffer | string) => {
+			if (data instanceof String) {
+				data = Buffer.from(data);
+			}
+
+			fileBuffer = Buffer.concat([fileBuffer, data as Buffer]);
+		});
```

Comment:
By changing this type check you can remove the need for `as`

```suggestion
		part.on('data', (data: Buffer | string) => {
			if (typeof data === 'string') {
				data = Buffer.from(data);
			}

			fileBuffer = Buffer.concat([fileBuffer, data]);
		});
```